openai/logger.py

```diff
@@ -0,0 +1,277 @@
+try:
+    import wandb
+
+    WANDB_AVAILABLE = True
+except:
+    WANDB_AVAILABLE = False
+
+
+if WANDB_AVAILABLE:
+    import datetime
+    import io
+    import json
+    from pathlib import Path
+
+    import numpy as np
+    import pandas as pd
+
+    from openai import File, FineTune
+
+
+class Logger:
+    """
+    Log fine-tune jobs to Weights & Biases
+    """
+
+    if not WANDB_AVAILABLE:
+        print("Logging requires wandb to be installed. Run `pip install wandb`.")
+    else:
+        _wandb_api = None
+        _logged_in = False
+
+    @classmethod
+    def sync(
+        cls,
+        id=None,
+        n_jobs=None,
+        project="GPT-3",
+        entity=None,
+        force=False,
+        **kwargs_wandb_init,
+    ):
+        """
+        Sync fine-tune job to Weights & Biases.
+        :param id: The id of the fine-tune job (optional)
+        :param n_jobs: Number of most recent fine-tune jobs to log when an id is not provided
+        :param project: Name of the project where you're sending runs. By default, it is "GPT-3".
+        :param entity: Username or team name where you're sending runs. By default, your default entity is used, which is usually your username.
+        :param force: Forces logging and overwrite existing wandb run of the same finetune job.
+        """
+
+        if not WANDB_AVAILABLE:
+            return
+
+        if id:
+            fine_tune = FineTune.retrieve(id=id)
+            fine_tune.pop("events", None)
+            fine_tunes = [fine_tune]
+
+        else:
+            # get list of fine_tune to log
+            fine_tunes = FineTune.list()
+            if not fine_tunes or fine_tunes.get("data") is None:
+                print("No fine-tune jobs have been retrieved")
+                return
+            fine_tunes = fine_tunes["data"][-n_jobs if n_jobs is not None else None :]
+
+        # log starting from oldest fine_tune
+        show_warnings = False if id is None and n_jobs is None else True
```

Comment:
Nit: Either [1] add a comment to briefly explain the `show_warnings` logic here and in L81 ("Show individual warnings if the user specifies a fine-tune or a specific number of fine-tunes, otherwise only warn if there are no new successful fine-tunes to sync."), or [2] rename this to `show_individual_warnings`. I think it'll make the warning logic slightly easier to grok quickly.

Reply:
Good idea!